handler/core/src/main/java/io/knotx/fragments/handler/action/InlinePayloadActionFactory.java

```diff
@@ -47,23 +57,38 @@ public Action create(String alias, JsonObject config, Vertx vertx, Action doActi
     if (doAction != null) {
       throw new IllegalArgumentException("Inline Payload Action does not support doAction");
     }
-    if (!config.containsKey("payload")) {
+    if (!config.containsKey(PAYLOAD_KEY)) {
       throw new IllegalArgumentException("Inline Payload Action requires payload parameter");
     }
     return (fragmentContext, resultHandler) -> {
-      String key = config.getString("alias", alias);
-      Object payload = config.getMap().get("payload");
+      ActionLogger actionLogger = createLogger(alias, config);
+      String key = config.getString(ALIAS_KEY, alias);
+      Object payload = config.getMap().get(PAYLOAD_KEY);
 
       Future<FragmentResult> resultFuture = Future
-          .succeededFuture(toResult(fragmentContext, key, payload));
+          .succeededFuture(toResult(fragmentContext, key, payload, actionLogger));
       resultFuture.setHandler(resultHandler);
     };
   }
 
-  private FragmentResult toResult(FragmentContext fragmentContext, String key, Object payload) {
+  private FragmentResult toResult(FragmentContext fragmentContext, String key, Object payload,
+      ActionLogger actionLogger) {
     Fragment fragment = fragmentContext.getFragment();
     fragment.appendPayload(key, payload);
```

Comment:
Can we move those two lines:
```
fragment.appendPayload(key, payload);
logSubstitution(actionLogger, key, payload);
```
to dedicated method `updatePayload` and call it from here:
```
return (fragmentContext, resultHandler) -> {
      ActionLogger actionLogger = createLogger(alias, config);
      String key = config.getString(ALIAS_KEY, alias);
      Object payload = config.getMap().get(PAYLOAD_KEY);
      updatePayload(...);
      ...
};
```